src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1242,19 +1246,192 @@
+                                    {:keys [type constraints fact-binding args]} condition
+                                    cmeta (meta condition)]
+                                (handle-expr prev
+                                             (with-meta (compile-condition
+                                                          type (first args) constraints
+                                                          fact-binding env)
+                                                        ;; Remove all metadata but file and line number
+                                                        ;; to protect from evaluating usafe metadata
+                                                        ;; See PR 243 for more detailed discussion
+                                                        (select-keys cmeta [:line :file]))
+                                             id
+                                             :alpha-expr
+                                             {:file (or (:file cmeta) *file*)
+                                              :compile-ctx {:condition condition
+                                                            :env env
+                                                            :msg "compiling alpha node"}})))
+                            {}
+                            alpha-graph)
+        exprs (reduce-kv (fn [prev id production-node]
+                             (let [production (-> production-node :production)]
+                               (handle-expr prev
+                                            (with-meta (compile-action (:bindings production-node)
+                                                                       (:rhs production)
+                                                                       (:env production))
+                                                       (meta (:rhs production)))
+                                            id
+                                            :action-expr
+                                            ;; ProductionNode expressions can be arbitrary code, therefore we need the
+                                            ;; ns where the production was define so that we can compile the expression
+                                            ;; later.
+                                            {:ns (:ns-name production)
+                                             :file (-> production :rhs meta :file)
+                                             :compile-ctx {:production production
+                                                           :msg "compiling production node"}})))
+                           alpha-exprs
+                           (:id-to-production-node beta-graph))]
+    (reduce-kv
+      (fn [prev id beta-node]
+        (let [condition (:condition beta-node)
+              condition (if (symbol? condition)
+                          (.loadClass (clojure.lang.RT/makeClassLoader) (name condition))
```

Comment:
Assuming that anything that is a symbol is a class seems likely to be a bug - we'll need to test it though.  Looks like there is some [existing logic along these lines](https://github.com/cerner/clara-rules/blob/0.18.0/src/main/clojure/clara/rules/compiler.clj#L181).  Also will we enter this path from ClojureScript at all?

Reply:
This is previous functionality, just shuffled around to work in my new approach. It does seem a little questionable, but i assumed that maintaining the old way would be better than changing functionality.

Reply:
Yeah, it should be fine to leave as is with regards to this issue.  I have a suspicion there is a way to break this (as in make it throw ClassNotFoundException) but I don't think this change would make it any more broken since the assumption is the same before and after.  We can look at this separately.

Reply:
I think `resolve` would be more appropriate, however, `resolve` has a defect that can throw CNFE as well.
https://dev.clojure.org/jira/browse/CLJ-1403

this may be fixed in 1.10, but that is future thinking.

Even if `resolve` returns `nil` as expected here, handling the nil/exception case would be nicer than letting it bubble up without much context.

I agree though that this is not related to the work of this PR.